stan/math/rev/core/var.hpp

```diff
@@ -181,6 +184,56 @@ namespace stan {
       // NOLINTNEXTLINE
       var(unsigned long x) : vi_(new vari(static_cast<double>(x))) { }  // NOLINT
 
+      /**
+       * Construct a variable from the specified arithmetic argument
+       * by constructing a new <code>vari</code> with the argument
+       * cast to <code>double</code>, and a zero adjoint. Only works
+       * if the imaginary part is zero.
+       *
+       * @param x Value of the variable.
+       */
+      explicit var(std::complex<double> x) {
+        assert(imag(x) == 0);
```

Comment:
This should be an exception.  Although users of the Stan language won't explicitly call it, it's going to be a constructor in the math library. 

I don't know how much printing for complex is built in, but maybe
```
if (imag(x) == 0) {
  std::stringstream ss;
  ss << "Imaginary part of std::complex used to construct var must be zero."
    << " Found argument = " << x << std::endl;
  std::string msg = ss.str();
  throw std::invalid_argument(msg);
}
```